src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1296 @@
+            // {
+            //     reads_complete += 1;
+            // }
+            // if rlen == 0
+            // {
+            //     break;
+            // }
+
+            base_idx += self.ibs;
+        }
+
+        buf.truncate(base_idx);
+        Ok(ReadStat {
+            reads_complete,
+            reads_partial,
+            records_truncated: 0,
+        })
+    }
+
+    /// Force-fills a buffer, ignoring zero-length reads which would otherwise be
+    /// interpreted as EOF.
+    /// Note: This will not return unless the source (eventually) produces
+    /// enough bytes to meet target_len.
+    fn force_fill(&mut self, buf: &mut [u8], target_len: usize) -> Result<usize, Box<dyn Error>> {
+        let mut base_idx = 0;
+        while base_idx < target_len {
+            base_idx += self.read(&mut buf[base_idx..target_len])?;
+        }
+
+        Ok(base_idx)
+    }
+}
+
+struct Output<W: Write> {
+    dst: W,
+    obs: usize,
+    cflags: OConvFlags,
+}
+
+impl Output<io::Stdout> {
+    fn new(matches: &Matches) -> Result<Self, Box<dyn Error>> {
+        let obs = parseargs::parse_obs(matches)?;
+        let cflags = parseargs::parse_conv_flag_output(matches)?;
+
+        let dst = io::stdout();
+
+        Ok(Output { dst, obs, cflags })
+    }
+
+    fn fsync(&mut self) -> io::Result<()> {
+        self.dst.flush()
+    }
+
+    fn fdatasync(&mut self) -> io::Result<()> {
+        self.dst.flush()
+    }
+}
+
+#[cfg(target_os = "linux")]
+fn make_linux_oflags(oflags: &OFlags) -> Option<libc::c_int> {
+    let mut flag = 0;
+
+    // oflag=FLAG
+    if oflags.append {
+        flag |= libc::O_APPEND;
+    }
+    if oflags.direct {
+        flag |= libc::O_DIRECT;
+    }
+    if oflags.directory {
+        flag |= libc::O_DIRECTORY;
+    }
+    if oflags.dsync {
+        flag |= libc::O_DSYNC;
+    }
+    if oflags.noatime {
+        flag |= libc::O_NOATIME;
+    }
+    if oflags.noctty {
+        flag |= libc::O_NOCTTY;
+    }
+    if oflags.nofollow {
+        flag |= libc::O_NOFOLLOW;
+    }
+    if oflags.nonblock {
+        flag |= libc::O_NONBLOCK;
+    }
+    if oflags.sync {
+        flag |= libc::O_SYNC;
+    }
+
+    if flag != 0 {
+        Some(flag)
+    } else {
+        None
+    }
+}
+
+impl Output<File> {
+    fn new(matches: &Matches) -> Result<Self, Box<dyn Error>> {
+        fn open_dst(
+            path: &Path,
+            cflags: &OConvFlags,
+            oflags: &OFlags,
+        ) -> Result<File, Box<dyn Error>> {
+            let mut opts = OpenOptions::new();
+            opts.write(true)
+                .create(!cflags.nocreat)
+                .truncate(!cflags.notrunc)
+                .create_new(cflags.excl)
+                .append(oflags.append);
+
+            #[cfg(target_os = "linux")]
+            if let Some(libc_flags) = make_linux_oflags(oflags) {
+                opts.custom_flags(libc_flags);
+            }
+
+            let dst = opts.open(path)?;
+            Ok(dst)
```

Comment:
```suggestion
            opts.open(path)
```

Reply:
This was actually due to a compiler message that went away when I did it this way. Addressed the message, and made this change.